internal/providers/terraform/parser.go

```diff
@@ -78,21 +78,18 @@ func (p *Parser) createResource(d *schema.ResourceData, u *schema.UsageData) *sc
 	}
 }
 
-func (p *Parser) parseJSON(j []byte, usage map[string]*schema.UsageData) ([]*schema.Resource, error) {
-	resources := p.loadUsageFileResources(usage)
-
-	if !gjson.ValidBytes(j) {
-		return resources, errors.New("invalid JSON")
-	}
-
-	parsed := gjson.ParseBytes(j)
-	providerConf := parsed.Get("configuration.provider_config")
-	conf := parsed.Get("configuration.root_module")
-	vars := parsed.Get("variables")
-
-	vals := parsed.Get("planned_values.root_module")
-	if !vals.Exists() {
-		vals = parsed.Get("values.root_module")
+func (p *Parser) parseJSONResources(parseExisting bool, baseResources []*schema.Resource, usage map[string]*schema.UsageData, parsed, providerConf, conf, vars gjson.Result) []*schema.Resource {
+	var resources []*schema.Resource
+	resources = append(resources, baseResources...)
+	var vals gjson.Result
+	if parseExisting {
+		vals = parsed.Get("prior_state.values.root_module")
+		// TODO: Does the key differ in state file?
+	} else {
+		vals = parsed.Get("planned_values.root_module")
+		if !vals.Exists() {
+			vals = parsed.Get("values.root_module")
+		}
 	}
```

Comment:
> // TODO: Does the key differ in state file?

Yes the `values.root_module` is used in the state file. So should be like this:

```suggestion
	if parseExisting {
		vals = parsed.Get("prior_state.values.root_module")
		if !vals.Exists() {
			vals = parsed.Get("values.root_module")
		}
	} else {
		vals = parsed.Get("planned_values.root_module")
	}
```